### PR TITLE
Added ability for views to have a type

### PR DESCRIPTION
**Pull request by Blackbaud-TrevorBurch**

### CODE REVIEW

File: src/app/public/modules/list/fixtures/list-view-test-type.component.fixture.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
+//#region imports
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/list/fixtures/list-view-test-type.component.fixture.ts]: These collapsible region comments are no longer needed - VS Code will automatically collapse comments.

File: src/app/public/modules/list/list.component.spec.ts [hunk 2]
```diff
@@ -1013,8 +1065,10 @@ describe('List Component', () => {
         expect(component.list.views.length).toBe(2);
         expect(component.list.views[0] instanceof ListViewTestComponent).toBeTruthy();
         expect(component.list.views[0].label).toBe('First');
-        expect(component.list.views[1] instanceof ListViewTestComponent).toBeTruthy();
+        expect(component.list.views[0].type).toBe(SkyListViewType.Custom);
+        expect(component.list.views[1] instanceof ListViewTestTypeComponent).toBeTruthy();
         expect(component.list.views[1].label).toBe('Second');
+        expect(component.list.views[1].type).toBe(SkyListViewType.Grid);
```
Comment (Blackbaud-AlexKingman) [inline src/app/public/modules/list/list.component.spec.ts]: This certainly covers the new code, but I feel like we just tacked onto another test here. Should we have a separate test that's deliberately testing for "should default to SkyListViewType.Custom" and "should be able to assign SkyListViewTypes via the constructor" ?.

File: src/app/public/modules/list/fixtures/list-view-test-type.component.fixture.ts (hunk 1 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/list/fixtures/list-view-test-type.component.fixture.ts]: Removed. That was just from me copying in the old component that I based this off of.

File: src/app/public/modules/list/list.component.spec.ts (hunk 2 above)
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/list/list.component.spec.ts]: Added a couple tests

**Pull request closed without merging**